core/chains/cosmos/cosmostxm/helpers_test.go

```diff
@@ -28,3 +29,16 @@ func (txm *Txm) MarshalMsg(msg sdk.Msg) (string, []byte, error) {
 func (txm *Txm) SendMsgBatch(ctx context.Context) {
 	txm.sendMsgBatch(ctx)
 }
+
+func (ka *KeystoreAdapter) Accounts(ctx context.Context) ([]string, error) {
+	ka.mutex.Lock()
+	err := ka.updateMappingLocked()
+	if err != nil {
+		ka.mutex.Unlock()
+		return nil, err
+	}
+	addresses := maps.Keys(ka.addressToPubKey)
+	ka.mutex.Unlock()
```

Comment:
nit/ could defer unlock here for simplicity:
```suggestion
	defer ka.mutex.Unlock()
	err := ka.updateMappingLocked()
	if err != nil {
		return nil, err
	}
	addresses := maps.Keys(ka.addressToPubKey)
```

Reply:
+1